core/src/main/java/com/datastrato/gravitino/storage/relational/service/CatalogMetaService.java

```diff
@@ -55,6 +55,17 @@ public CatalogPO getCatalogPOByMetalakeIdAndName(Long metalakeId, String catalog
     return catalogPO;
   }
 
+  // Catalog may be deleted, so the CatalogPO may be null.
+  // We don't choose to delete the securable object when we delete the catalog,
+  // Because they will bring conflicts if we are updating securable object
+  public CatalogPO getCatalogPOById(Long catalogId) {
```

Comment:
You should add the nullable annotation, also the explanation should be in the right place, not here, it's a bit weird added here.